docker.go

```diff
@@ -119,8 +124,51 @@ type (
 	TagStruct struct {
 		Tag string `json:"Tag"`
 	}
+
+	LayerStatus struct {
+		Status string
+		Time   float64 // Time in seconds; only set for DONE layers
+	}
+
+	CacheMetrics struct {
+		TotalLayers int                 `json:"total_layers"`
+		Done        int                 `json:"done"`
+		Cached      int                 `json:"cached"`
+		Errored     int                 `json:"errored"`
+		Cancelled   int                 `json:"cancelled"`
+		Layers      map[int]LayerStatus `json:"layers"`
+	}
+
+	tee struct {
+		w      io.Writer
+		status chan string
+	}
 )
 
+func (t *tee) Write(p []byte) (n int, err error) {
```

Comment:
Add comments in this method to understand the data flow. First one going to stdout and second one going to the channel

Reply:
Done